File: scripts/auto-post-news.py
#!/usr/bin/env python3
"""
Automated News Posting Script for momwatching.com
- Searches for education news
- Generates SEO-optimized content using AI
- Fetches images from Pixabay
- Updates news database and pushes to Git
"""

import os
import sys
import json
import time
import requests
from datetime import datetime
from pathlib import Path
import subprocess
import random

# Add parent directory to path
SCRIPT_DIR = Path(__file__).parent
PROJECT_ROOT = SCRIPT_DIR.parent

# Configuration
PIXABAY_API_KEY = "46430210-e974faa2be384e27927d08033"
NEWS_KEYWORDS = ["입시", "고등학교", "대입", "수시", "정시", "서울대", "진학", "학생부"]
NEWS_SOURCES = [
    {"name": "베리타스알파", "url": "https://www.veritas-a.com/"},
    {"name": "연합뉴스", "url": "https://www.yna.co.kr/"}
]

def search_news_topics():
    """Search for trending education news topics"""
    print("🔍 Searching for education news topics...")
    
    # Select random keyword
    keyword = random.choice(NEWS_KEYWORDS)
    
    # In production, this would:
    # 1. Search Brave API for recent news
    # 2. Scrape 베리타스알파 RSS feed
    # 3. Check 연합뉴스 education section
    
    print(f"✅ Selected topic: {keyword}")
    return keyword

def generate_article_with_ai(keyword):
    """Generate article content using AI (Claude/GPT)"""
    print(f"✍️  Generating article for: {keyword}")
    
    # Article generation prompt
    prompt = f"""다음 키워드로 고등학교 입시 뉴스 기사를 작성해주세요: {keyword}

요구사항:
1. 제목: 60자 이내, SEO 최적화 (예: "2026 {keyword} 정책 변화 - 고등학교 입시 전략")
2. 요약: 150-200자, 핵심 내용 + "momwatching.com에서 확인"
3. 본문: 600-800자, 학생/학부모가 알아야 할 실질적 정보
4. 톤: 전문적이면서 친근하게
5. 키워드 자연스럽게 포함: {keyword}, 입시, 고등학교

실제 뉴스처럼 작성하되, 구체적인 날짜나 출처는 일반적으로 표현해주세요."""

    # In production, use OpenAI/Anthropic API
    # For now, create template
    timestamp = datetime.now()
    
    article = {
        "id": f"news-{int(timestamp.timestamp())}",
        "title": f"2026 {keyword} 주요 변화 - 고등학교 입시 뉴스",
        "slug": f"{keyword}-update-{timestamp.strftime('%Y%m%d')}",
        "date": timestamp.isoformat(),
        "summary": f"최근 {keyword} 관련 중요한 변화가 있었습니다. 학생과 학부모가 꼭 알아야 할 핵심 내용을 정리했습니다. 자세한 입시 정보는 momwatching.com에서 확인하세요.",
        "content": f"""## {keyword} 최신 동향

최근 교육계에서 {keyword}와 관련된 중요한 변화가 발표되어 학생과 학부모들의 관심이 집중되고 있습니다.

### 주요 변화 내용

이번 발표의 핵심 내용은 다음과 같습니다:

**1. 정책 방향**
{keyword}와 관련하여 새로운 정책 방향이 제시되었습니다. 이는 학생들의 진로 선택과 입시 전략에 중요한 영향을 미칠 것으로 예상됩니다.

**2. 적용 시기**
2026학년도부터 단계적으로 적용될 예정이며, 현재 중학생과 고등학생 모두 주의깊게 살펴볼 필요가 있습니다.

**3. 영향 범위**
전국 고등학교의 입시 전략과 대학 진학 준비 방식에 전반적인 변화가 예상됩니다.

### 학생·학부모 대응 전략

전문가들은 다음과 같은 준비가 필요하다고 조언합니다:

- **정확한 정보 파악**: 변화된 정책을 정확히 이해하고 우리 아이에게 맞는 전략 수립
- **학교 선택**: 각 학교의 {keyword} 대응 전략과 진학 실적 비교 분석
- **전문가 상담**: 입시 전문가와 상담하여 맞춤형 진학 로드맵 작성

### 고등학교 선택이 중요한 이유

{keyword} 정책 변화로 인해 고등학교 선택의 중요성이 더욱 커졌습니다. 각 학교의 서울대 진학 실적, 수시/정시 비율, 교육 프로그램을 꼼꼼히 비교해보세요.

momwatching.com에서는 전국 고등학교의 서울대 진학 실적 순위와 상세 정보를 제공하고 있습니다. 학교별 5개년 추이, 지역별 비교 등 다양한 데이터를 확인하실 수 있습니다.""",
        "image": "/news/images/default.jpg",
        "keywords": [keyword, "입시", "고등학교", "진학", "2026"],
        "source": random.choice(NEWS_SOURCES)["name"]
    }
    
    print(f"✅ Article generated: {article['title']}")
    return article

def fetch_pixabay_image(keyword):
    """Fetch relevant image from Pixabay"""
    print(f"🖼️  Fetching image from Pixabay...")
    
    try:
        # Use generic education-related search terms
        search_terms = ["education", "students studying", "school", "learning"]
        search_term = random.choice(search_terms)
        
        url = f"https://pixabay.com/api/"
        params = {
            "key": PIXABAY_API_KEY,
            "q": search_term,
            "image_type": "photo",
            "per_page": 20,
            "safesearch": "true",
            "orientation": "horizontal"
        }
        
        response = requests.get(url, params=params, timeout=10)
        data = response.json()
        
        if data.get("hits"):
            image = random.choice(data["hits"][:10])  # Pick from top 10
            image_url = image["webformatURL"]
            print(f"✅ Image found: {image_url}")
            return image_url
        else:
            print("⚠️  No images found, using default")
            
    except Exception as e:
        print(f"❌ Error fetching image: {e}")
    
    return "/news/images/default.jpg"

def update_news_database(article):
    """Update news.json with new article"""
    print("💾 Updating news database...")
    
    news_file = PROJECT_ROOT / "data" / "news.json"
    
    # Read existing articles
    articles = []
    if news_file.exists():
        with open(news_file, 'r', encoding='utf-8') as f:
            articles = json.load(f)
    
    # Add new article at the beginning
    articles.insert(0, article)
    
    # Keep only last 100 articles
    articles = articles[:100]
    
    # Write back
    with open(news_file, 'w', encoding='utf-8') as f:
        json.dump(articles, f, ensure_ascii=False, indent=2)
    
    print(f"✅ Database updated ({len(articles)} total articles)")

def commit_and_push(article_title):
    """Commit changes and push to Git"""
    print("📤 Committing and pushing to Git...")
    
    try:
        os.chdir(PROJECT_ROOT)
        
        # Git operations
        subprocess.run(["git", "add", "data/news.json"], check=True)
        subprocess.run([
            "git", "commit", "-m", f"자동 뉴스 업데이트: {article_title}"
        ], check=True)
        subprocess.run(["git", "push", "origin", "main"], check=True)
        
        print("✅ Successfully pushed to Git")
        print("🚀 Vercel will auto-deploy the changes")
        
    except subprocess.CalledProcessError as e:
        print(f"❌ Git error: {e}")
        raise

def main():
    """Main execution flow"""
    print("=" * 60)
    print("🤖 Automated News Posting for momwatching.com")
    print(f"📅 {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    print("=" * 60)
    print()
    
    try:
        # 1. Search for news topic
        keyword = search_news_topics()
        
        # 2. Generate article with AI
        article = generate_article_with_ai(keyword)
        
        # 3. Fetch image
        image_url = fetch_pixabay_image(keyword)
        article["image"] = image_url
        
        # 4. Update database
        update_news_database(article)
        
        # 5. Commit and push
        commit_and_push(article["title"])
        
        print()
        print("=" * 60)
        print("✅ NEWS POSTING COMPLETED SUCCESSFULLY!")
        print(f"📝 Title: {article['title']}")
        print(f"🔗 Slug: {article['slug']}")
        print(f"📅 Date: {article['date']}")
        print("=" * 60)
        
        return 0
        
    except Exception as e:
        print()
        print("=" * 60)
        print(f"❌ ERROR: {e}")
        print("=" * 60)
        return 1

if __name__ == "__main__":
    sys.exit(main())

File: scripts/auto-post-news.js
#!/usr/bin/env node

const fs = require('fs').promises;
const path = require('path');
const { execSync } = require('child_process');

// Configuration
const PIXABAY_API_KEY = '46430210-e974faa2be384e27927d08033';
const NEWS_KEYWORDS = ['입시', '고등학교', '대입', '수시', '정시', '서울대', '진학'];
const NEWS_SOURCES = [
  { name: '베리타스알파', url: 'https://www.veritas-a.com/' },
  { name: '연합뉴스', url: 'https://www.yna.co.kr/' }
];

/**
 * Fetch news from web sources using web_search
 */
async function fetchNews() {
  console.log('🔍 Searching for education news...');
  
  // For now, we'll create a template. In production, this would use web scraping
  // or RSS feeds from 베리타스알파 and 연합뉴스
  
  const keyword = NEWS_KEYWORDS[Math.floor(Math.random() * NEWS_KEYWORDS.length)];
  const searchQuery = `${keyword} 뉴스 2026`;
  
  console.log(`Search query: ${searchQuery}`);
  
  // In a real implementation, you would:
  // 1. Use web_search tool or RSS feed parser
  // 2. Scrape article content
  // 3. Filter by relevance
  
  return {
    keyword,
    searchQuery,
    foundArticles: [] // Placeholder for actual scraped articles
  };
}

/**
 * Generate article using GPT (mock for now - would use OpenAI API)
 */
async function generateArticle(keyword) {
  console.log(`✍️  Generating article for keyword: ${keyword}`);
  
  const timestamp = new Date().toISOString();
  const slug = `${keyword}-news-${Date.now()}`;
  
  // Template article (in production, use GPT API)
  const article = {
    id: `news-${Date.now()}`,
    title: `2026 ${keyword} 최신 동향 - 고등학교 입시 뉴스`,
    slug: slug,
    date: timestamp,
    summary: `최근 ${keyword} 관련 주요 변화가 발표되었습니다. 학생과 학부모들이 알아야 할 핵심 내용을 정리했습니다. 자세한 입시 전략은 momwatching.com에서 확인하세요.`,
    content: `# ${keyword} 주요 변화 내용

최근 교육부에서 발표한 ${keyword} 관련 정책 변화가 학생과 학부모들의 큰 관심을 받고 있습니다.

## 주요 내용

이번 발표의 핵심은 다음과 같습니다:

1. **정책 방향**: ${keyword}와 관련된 새로운 방향성이 제시되었습니다.
2. **적용 시기**: 2026학년도부터 단계적으로 적용될 예정입니다.
3. **영향 범위**: 전국 고등학교와 대학 입시 전반에 영향을 미칠 것으로 보입니다.

## 학생 및 학부모 대응 방안

전문가들은 다음과 같은 대응이 필요하다고 조언합니다:

- 변화된 정책을 정확히 이해하고 준비
- 학교별 입시 실적과 전략 비교 분석
- 전문가 상담을 통한 맞춤형 진학 설계

## 더 알아보기

전국 고등학교의 서울대 진학 실적과 상세한 입시 정보는 momwatching.com에서 확인하실 수 있습니다.`,
    image: '/news/images/default.jpg',
    keywords: [keyword, '입시', '고등학교', '2026'],
    source: '베리타스알파'
  };
  
  return article;
}

/**
 * Fetch image from Pixabay
 */
async function fetchImage(keyword) {
  console.log(`🖼️  Fetching image for: ${keyword}`);
  
  try {
    const searchTerm = 'education students study'; // Generic education image
    const url = `https://pixabay.com/api/?key=${PIXABAY_API_KEY}&q=${encodeURIComponent(searchTerm)}&image_type=photo&per_page=3&safesearch=true`;
    
    const response = await fetch(url);
    const data = await response.json();
    
    if (data.hits && data.hits.length > 0) {
      const image = data.hits[0];
      console.log(`✅ Found image: ${image.pageURL}`);
      return image.webformatURL;
    }
  } catch (error) {
    console.error('Error fetching image from Pixabay:', error);
  }
  
  return '/news/images/default.jpg';
}

/**
 * Update news.json
 */
async function updateNewsDatabase(article) {
  console.log('💾 Updating news database...');
  
  const newsFilePath = path.join(__dirname, '..', 'data', 'news.json');
  
  let articles = [];
  try {
    const content = await fs.readFile(newsFilePath, 'utf8');
    articles = JSON.parse(content);
  } catch (error) {
    console.log('Creating new news.json file...');
  }
  
  articles.unshift(article);
  
  // Keep only last 100 articles
  if (articles.length > 100) {
    articles = articles.slice(0, 100);
  }
  
  await fs.writeFile(newsFilePath, JSON.stringify(articles, null, 2));
  console.log('✅ News database updated');
}

/**
 * Commit and push to Git
 */
function commitAndPush(articleTitle) {
  console.log('📤 Committing and pushing to Git...');
  
  try {
    const rootDir = path.join(__dirname, '..');
    
    execSync('git add data/news.json', { cwd: rootDir });
    execSync(`git commit -m "자동 뉴스 업데이트: ${articleTitle}"`, { cwd: rootDir });
    execSync('git push origin main', { cwd: rootDir });
    
    console.log('✅ Successfully pushed to Git');
    console.log('🚀 Vercel will auto-deploy the changes');
  } catch (error) {
    console.error('❌ Error pushing to Git:', error.message);
  }
}

/**
 * Main function
 */
async function main() {
  console.log('🤖 Starting automated news posting...');
  console.log(`📅 Time: ${new Date().toLocaleString('ko-KR')}`);
  console.log('─'.repeat(50));
  
  try {
    // 1. Fetch news
    const newsData = await fetchNews();
    
    // 2. Generate article
    const article = await generateArticle(newsData.keyword);
    
    // 3. Fetch image
    const imageUrl = await fetchImage(newsData.keyword);
    article.image = imageUrl;
    
    // 4. Update database
    await updateNewsDatabase(article);
    
    // 5. Git commit & push
    commitAndPush(article.title);
    
    console.log('─'.repeat(50));
    console.log('✅ News posting completed successfully!');
    console.log(`📝 Title: ${article.title}`);
    console.log(`🔗 Slug: ${article.slug}`);
  } catch (error) {
    console.error('❌ Error in main process:', error);
    process.exit(1);
  }
}

// Run if called directly
if (require.main === module) {
  main();
}

module.exports = { main };
